### ISSUE TITLE
W605 potential false positive with `\` in multi-line strings

### ISSUE THREAD

**Issue by AA-Turner** [charliermarsh/ruff#581]
Escaping a newline is explicitly allowed, see the table in Lexical Analysis: https://docs.python.org/3/reference/lexical_analysis.html#index-23

Reproducer:

```python
# bug.py
print("""\
Lorem ipsum dolar sit amet.
Lorem ipsum dolar \
sit amet.
Lorem ipsum dolar \
sit amet.""")
```

Sample output of `flake8` and `ruff`:

```doscon
I:\Development>type bug.py                         
print("""\                 
Lorem ipsum dolar sit amet.
Lorem ipsum dolar \        
sit amet.                  
Lorem ipsum dolar \        
sit amet.""")              

I:\Development>flake8 --isolated --select W bug.py 

I:\Development>ruff --select W bug.py              
Found 3 error(s).
'ug.py:1:10: W605 Invalid escape sequence: '\
'ug.py:3:20: W605 Invalid escape sequence: '\
'ug.py:5:20: W605 Invalid escape sequence: '\
```

I think there is also a bug in the error message here, as the first character is replaced by `'`.

Windows 10, Python 3.10

A

**Comment by charliermarsh** [comment]
Thank you! Will fix this today.

**Comment by charliermarsh** [comment]
I'm wondering if this is Windows-specific? I'm not able to repro on macOS. Might take me a bit as I'll need to setup a testing env for Windows.

**Comment by charliermarsh** [comment]
(I'm going through that process now, I need to be able to test on Windows anyway so this is a good reason to set it up.)

**Comment by AA-Turner** [comment]
> I'm wondering if this is Windows-specific?

Got it!

```bash
(venv) adam@Gyrostats:/mnt/i/Development$ file bug.py
bug.py: ASCII text, with CRLF line terminators
(venv) adam@Gyrostats:/mnt/i/Development$ ruff --select W bug.py
Found 3 error(s).
'ug.py:1:10: W605 Invalid escape sequence: '\
'ug.py:3:20: W605 Invalid escape sequence: '\
'ug.py:5:20: W605 Invalid escape sequence: '\
(venv) adam@Gyrostats:/mnt/i/Development$ file good.py
good.py: ASCII text
(venv) adam@Gyrostats:/mnt/i/Development$ ruff --select W good.py
(venv) adam@Gyrostats:/mnt/i/Development$
```

Incorrect parsing of carriage returns, it seems.

A

**Comment by charliermarsh** [comment]
I can reproduce on macOS by saving the life with CRLF newlines.

**Comment by AA-Turner** [comment]
Thanks for the quick fix!

A

**Comment by charliermarsh** [comment]
For sure! Thanks for the great Issue write-up. I'll cut a new release in a bit (want to include a few more small things).
